pandas/io/formats/style.py

```diff
@@ -1393,26 +1461,32 @@ def _background_gradient(
         text_color_threshold: float = 0.408,
         vmin: Optional[float] = None,
         vmax: Optional[float] = None,
+        gmap: Optional[Sequence] = None,
     ):
         """
-        Color background in a range according to the data.
+        Color background in a range according to the data or a gradient map
         """
-        if (
-            not isinstance(text_color_threshold, (float, int))
-            or not 0 <= text_color_threshold <= 1
-        ):
-            msg = "`text_color_threshold` must be a value from 0 to 1."
-            raise ValueError(msg)
-
+        if gmap is None:
+            gmap = s.to_numpy(dtype=float)
+        else:
+            if s.ndim == 1:  # s is Series(n): gmap can be (n,), (n,1) or (1,n)
+                gmap = np.asarray(gmap, dtype=float).reshape(-1)
+            else:  # s is DataFrame(n,m): gmap must be (n,m)
```

Comment:
if this IS a DataFrame then it should be aligned to the styled frame no? does a DataFrame make sense here in any event?

Reply:
I don't think you are testing a DataFrame. Can you enumerate when that is useful? is only a 1D useful?

Reply:
I had a (n,m) array test but not specifcally a DataFrame. I have added this. I think users could find many interesting uses for this. As well as quickly inverting the color map described above with `gmap=-df. Here is an example of highlighting a matrix based on parameters outside of the matrix , in this case covariance being highlighted by correlation. And an example of highlighting based on a function dependent upon the data. 

<img width="932" alt="Screenshot 2021-03-04 at 10 56 06" src="https://user-images.githubusercontent.com/24256554/109946743-dc8de900-7cd8-11eb-92bd-7192310a6855.png">

Reply:
here's the deal if you have a Series/DataFrame then you *must* align. I fyou have an array it *must* be the exact shape as the existing.

Let's start simple and only select 1D for now. No need to go crazy here.

Reply:
I'm sorry but I don't understand the requested change. This method already works if `axis` is `None`, i.e. on a DataFrame (or a `subset` of one). In the case `axis` is `None` are you then suggesting prohibiting the `gmap` functionality, so that it only works when `axis` is in `[0, 1]`, i.e. only on Series??

Or rather is it the case that my code only deals with `gmap` as arrays, (or arrays converted directly from Series/DataFrame objects), and doesn't address aligning the indexes of those objects with the underlying data (and possibly addressing the `subset` issue as well)?

I hope you mean the latter, since I think, restricting this functionality only to 1D, will be a shame/rest and probably more confusing the user who want to use it. And otherwise I can address the alignment/reindexing issue.

Reply:
Ah ok after thinking about it I think you do mean the latter. I think you expect this to work:
![Screen Shot 2021-03-09 at 10 33 53](https://user-images.githubusercontent.com/24256554/110450107-30c30f80-80c3-11eb-878c-2ca8a9bc3fb5.png)
This didn't work previously, as you highlighted, but now with the alignment it does - yes this is much better!

Tests for this added.